### PR TITLE
Dev 1377/pyraphtory jvmtoml

### PR DESCRIPTION
**Pull request by ljeub-pometry**
### What changes were proposed in this pull request?

Reworked the build system for pyraphtory such that pip install works from GitHub and builds core jars to put into source distribution and jre and dependencies are included in wheel. Also changed it such that JAVA_HOME is ignored unless the special environment variable PYRAPHTORY_USE_SYSTEM_JAVA is set to something truthy

### Why are the changes needed?

Make it easier to install/develop pyraphtory

### Does this PR introduce any user-facing change? If yes is this documented?

Only for developers

### How was this patch tested?

The tests pass after refactoring. Also confirmed that the generated sdist can be installed and results in a working pyraphtory on a barebones linux vm without sbt or java installed.

### Are there any further changes required?

ivy dependency conflict resolution could use some more cleanup. The current solution is a hack.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,84 +2,89 @@ SHELL:=/bin/bash -euxo pipefail
...
+.PHONY: sbt-thin-build
+sbt-thin-build: version clean sbt-build
+
+
+.PHONY: python-build
+python-build: version
+	python -m pip install .
+
+
+.PHONY: python-build-editable
```
Comment (miratepuffin) [inline Makefile:55]: 🙌🏻 🙌🏻 🙌🏻

File: build.sbt [hunk 2]
```diff
@@ -128,7 +128,8 @@ lazy val arrowCore =
                     chronicleMap,
                     fastUtil,
                     commonsLang,
-                    junitInterface
+                    junitInterface,
+                    netty classifier osDetectorClassifier.value,
```
Comment (miratepuffin) [inline build.sbt:132]: Oh nice! Does this fix the windows issue by chance?

File: examples/lotr/src/main/python/LOTR-PyRaphtory.ipynb [hunk 3]
```diff
@@ -2,53 +2,118 @@
...
      "name": "#%%\n"
     }
-   }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Requirement already satisfied: pyvis in /Users/lucasjeub/opt/miniconda3/lib/python3.10/site-packages (0.3.0)\n",
```
Comment (miratepuffin) [inline examples/lotr/src/main/python/LOTR-PyRaphtory.ipynb:53]: Might want to clear this just as its got your name in it.

**Review by miratepuffin — APPROVED** [review]

**Review by miratepuffin — APPROVED** [review]

**Review by Haaroon — CHANGES_REQUESTED** [review]
if i understand correctly this merges the pyraphtory_jvm into pyraphtory, if so then 

- the docs need to be changed to point to the new way to install pyraphtory for users. 
- binder environment needs to be changed otherwise it will fail 

otherwise it looks good, i just built this on my mac

File: examples/lotr/src/main/python/LOTR-PyRaphtory.ipynb (hunk 3 above)
Comment (ljeub-pometry) [reply examples/lotr/src/main/python/LOTR-PyRaphtory.ipynb]: done

File: build.sbt (hunk 2 above)
Comment (ljeub-pometry) [reply build.sbt:132]: Not sure, it is something Fabian did to try to reduce the number of package version conflicts

**Comment by ljeub-pometry** [comment]
Fixed binder environment and removed mention of pyraphtory_jvm from the install instructions. We need to add more docs on advanced install (install from source, distributed option, etc)

**Comment by miratepuffin** [comment]
Will merge this for the time being and we can make a ticket for the advanced docs

**Review by Haaroon — APPROVED** [review]

**Pull request merged**